alma/controllers/hook/StateHookController.php

```diff
@@ -43,38 +43,84 @@ final class StateHookController extends AdminHookController
     use OrderDataTrait;
 
     /**
+     * Execute refund or trigger payment on change state
      * @param $params
      *
      * @throws PrestaShopDatabaseException
      * @throws PrestaShopException
      */
     public function run($params)
     {
-        if (!Settings::isRefundEnabledByState()) {
-            return;
-        }
-
         $order = new Order($params['id_order']);
         $newStatus = $params['newOrderStatus'];
         if ($order->module !== 'alma' || !$order_payment = $this->getOrderPaymentOrFail($order)) {
             return;
         }
-
+        $alma = ClientHelper::defaultInstance();
+        if (!$alma) {
+            return;
+        }
         $id_payment = $order_payment->transaction_id;
-        if ($newStatus->id == Settings::getRefundState()) {
-            $alma = ClientHelper::defaultInstance();
-            if (!$alma) {
+
+        switch ($newStatus->id) {
+            case Settings::getRefundState():
+                if (Settings::isRefundEnabledByState()) {
+                    Logger::instance()->info('[Alma] Option refund enabled');
+                    $this->refund($alma, $id_payment, $order);
+                }
+                Logger::instance()->info('[Alma] Option refund disabled');
```

Comment:
you should put `else` or `break` in `if` statement to log the `refund disabled` information,

actually: you are always write the `disabled log`

Reply:
you can maybe just log when `refund` is disabled and add some context about why log is written : 

```
[Alma] Option refund disabled: refund is not performed on %s refund state
```

(do the same for `paymentTrigger` 😉 )